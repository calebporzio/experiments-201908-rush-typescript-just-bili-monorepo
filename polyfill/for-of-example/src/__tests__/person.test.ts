import { default as whatever, Person, PersonLikeInterface } from '..'
import { people } from './__fixtures__'

describe('Person', () => {
  let subject: Person
  test('constructor', () => {
    const { name } = people[0] as PersonLikeInterface
    subject = new Person(name)
    expect(String(subject)).toBe(`Hello, my name is ${name}`)
  })

  test('default export', () => {
    // @ts-ignore
    const list = whatever(people)
    expect(list[0]).toMatchObject({ name: 'Luke Skywalker' })
    expect(list).toMatchSnapshot()
  })
})
