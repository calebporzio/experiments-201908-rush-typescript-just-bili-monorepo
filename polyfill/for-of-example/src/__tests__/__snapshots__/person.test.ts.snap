// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person default export 1`] = `
Array [
  Person {
    "lang": undefined,
    "name": "Luke Skywalker",
  },
  Person {
    "lang": undefined,
    "name": "Yoda Whatev",
  },
]
`;
